src/main/frontend/src/features/CoursePreview.js

```diff
@@ -1,41 +1,73 @@
 import ErrorAlert from './alerts/ErrorAlert';
+import Form from 'react-bootstrap/Form';
 import Header from './Header';
 import Image from 'react-bootstrap/Image';
 import InfoAlert from './alerts/InfoAlert';
 import Row from 'react-bootstrap/Row';
 
 function CoursePreview(props) {
 
+  // These values are needed to add the courses to the LMS.
+  const idToken = useSelector(selectIdToken);
+  const target = useSelector(selectTarget);
+  const state = useSelector(selectState);
+  const rootOutcomeGuid = useSelector(selectRootOutcomeGuid);
+  const selectedModules = useSelector(selectSelectedModules);
+
+  // Some courses may not have a valid cover image, use a default instead
+  const courseCoverUrl = parseCourseCoverImage(props.course.cover_img_url, true);
+  const dispatch = useDispatch();
+
   // In case there's an error adding the links, display an error message.
   const [errorAddingLinks, setErrorAddingLinks] = useState(false);
   const errorMessage = 'Oops, the Lumen content was not successfully added. Please try again or contact Lumen Support.';
 
-  const dispatch = useDispatch();
+  // The 'Select All' checkbox is controlled and depends on the state, enables or disables all the modules at the same time.
+  // If the course has not been paired with a Lumen course, 'Select All' must be checked.
+  const [selectAllChecked, setSelectAllChecked] = useState(rootOutcomeGuid === null);
+  // If the course has been paired with a Lumen course we must display a different text for the button.
+  const addButtonText = rootOutcomeGuid === null ? 'Add Course' : 'Add Content';
 
-  const resetSelectedCourse = () => {
+  // When the user clicks cancel it should reset the module and the course selection.
+  const resetSelection = () => {
     dispatch(changeSelectedCourse(null));
+    dispatch(toggleAllModules(0));
+  }
+
+  // When the user checks or unchecks 'Select All' it must toggle all the modules.
+  const handleSelectAll = () => {
+    const newSelectAll = !selectAllChecked;
+    setSelectAllChecked(newSelectAll);
+    dispatch(toggleAllModules(newSelectAll));
   }
 
-  // Some courses may not have a valid cover image, use a default instead
-  const courseCoverUrl = parseCourseCoverImage(props.course.cover_img_url);
   const addCourseToLMS = () => {
-      const ltiLaunchData = JSON.parse(document.getElementById('root').getAttribute('lti-launch-data'));
-      const idToken = ltiLaunchData.id_token;
-      const target = ltiLaunchData.target;
-      const state = ltiLaunchData.state;
-      const contextAPIUrl = target.replace("/lti3", "/context");
+
+      let contextAPIUrl = target.replace("/lti3", "/context");
+      contextAPIUrl += `?state=${state}`;
+      // Append the selected moduleIds to the PUT request as an additional parameter.
+      // Do not append moduleIds when all the modules have been selected.
+      if (Array.isArray(props.course.table_of_contents) && props.course.table_of_contents.length !== selectedModules.length) {
```

Comment:
Could you please send the `module_ids` in the body instead of as a request parameter?

Reply:
Also, I noticed that this seems to just collect the index of the module_id and send that rather than the actual module_id value sent from Harmony. Could you please have it generate an array of the module_ids as provided by Harmony?

Reply:
I'll check this, thanks!

Reply:
Now I send the 'module_ids' in the body as values separared by comma.

I still need to think about the moduleIds because I'm still sending the indexes.

Reply:
Hey @mgwozdz-unicon, now I'm sending the module_ids as an array to the backend in the request's body.